R/lazyframe-s3-base.R

```diff
@@ -27,3 +27,39 @@ head.polars_lazy_frame <- head.polars_data_frame
 
 #' @exportS3Method utils::tail
 tail.polars_lazy_frame <- tail.polars_data_frame
+
+
+# Try to match `tibble` behavior as much as possible, following
+# https://tibble.tidyverse.org/articles/invariants.html#column-subsetting
+# TODO: add document
+#' @export
+`[.polars_lazy_frame` <- function(x, i, j, ..., drop = FALSE) {
+  # get from `[.polars_data_frame`
+  i_arg <- substitute(i)
+  j_arg <- substitute(j) # nolint: object_usage_linter
+
+  if (isTRUE(drop)) {
+    warn(c(`!` = "`drop = TRUE` is not supported for LazyFrame."))
+  }
+  if (!missing(i)) {
+    n_real_args <- nargs() - !missing(drop)
+    if (n_real_args > 2) {
+      abort(
+        c(
+          `!` = "Cannot subset rows of a LazyFrame with `[`.",
+          i = "There are several functions that can be used to get a specific range of rows.",
+          `*` = "`$slice()` can be used to get a slice of rows.",
+          `*` = "`$gather_every()` can be used to take every nth row.",
+          `*` = "`$reverse()` can be used to reverse the order of rows."
+        )
```

Comment:
Technically, I believe that if `i` is `NULL`, `TRUE`, or `FALSE`, it needs to be allowed. (Related to #343)

Reply:
done